src/test/search.spec.js

```diff
@@ -0,0 +1,151 @@
+    expect(confirmArticle[0].authorId).to.equal(firstArticle.authorId);
+    expect(confirmTag[0].name).to.equal(firstTag.name);
+    expect(confirmArticleTag[0].articleId).to.equal(firstArticleTag.articleId);
+    expect(confirmArticleTag[0].tagId).to.equal(firstArticleTag.tagId);
+
+    const res = await server().get(
+      `${SEARCH_API}?search=${firstUser.username}`
+    );
+    expect(res.status).to.equal(200);
+    expect(res.body.data).to.be.an('array');
+    expect(res.body.data[0].username).to.equal(firstUser.username);
+    expect(res.body.data[0].title).to.equal(firstArticle.title);
+    expect(res.body.data[0].body).to.equal(firstArticle.body);
+    expect(res.body.data[0].description).to.equal(firstArticle.description);
+    expect(res.body.data.length).to.equal(1);
+    expect(res.body.data[0].user_score).to.be.within(0, 1);
+    expect(res.body.data[0].title_score).to.be.within(0, 1);
+    expect(res.body.data[0].tag_score).to.be.within(0, 1);
+  });
+
+  it('should search for articles by article title', async () => {
+    const firstUser = await createUser();
+    const firstArticle = await createArticle(
+      await generateArticle({ authorId: firstUser.id })
+    );
+    const firstTag = await createTag({ name: 'Bootcamp' });
+    const firstArticleTag = await createArticleTag({
+      tagId: firstTag.id,
+      articleId: firstArticle.id
+    });
+
+    const confirmAuthor = await BaseRepository.findItAll(db.User);
+    const confirmArticle = await BaseRepository.findItAll(db.Article);
+    const confirmTag = await BaseRepository.findItAll(db.Tags);
+    const confirmArticleTag = await BaseRepository.findItAll(db.ArticleTags);
+
+    expect(confirmAuthor[0].username).to.equal(firstUser.username);
+    expect(confirmArticle[0].authorId).to.equal(firstArticle.authorId);
+    expect(confirmTag[0].name).to.equal(firstTag.name);
+    expect(confirmArticleTag[0].articleId).to.equal(firstArticleTag.articleId);
+    expect(confirmArticleTag[0].tagId).to.equal(firstArticleTag.tagId);
+
+    const res = await server().get(
+      `${SEARCH_API}?search=${firstArticle.title}`
+    );
+    expect(res.status).to.equal(200);
+    expect(res.body.data).to.be.an('array');
+    expect(res.body.data[0].username).to.equal(firstUser.username);
+    expect(res.body.data[0].title).to.equal(firstArticle.title);
+    expect(res.body.data[0].body).to.equal(firstArticle.body);
+    expect(res.body.data[0].description).to.equal(firstArticle.description);
+    expect(res.body.data.length).to.equal(1);
+    expect(res.body.data[0].user_score).to.be.within(0, 1);
+    expect(res.body.data[0].title_score).to.be.within(0, 1);
+    expect(res.body.data[0].tag_score).to.be.within(0, 1);
+  });
+
+  it('should search for articles by tag', async () => {
+    const firstUser = await createUser();
+    const firstArticle = await createArticle(
+      await generateArticle({ authorId: firstUser.id })
```

Comment:
```

const article = await generateArticle({ authorId: firstUser.id });
article.title = 'bush gehs';
const secondarticle = await generateArticle({ authorId: firstUser.id });
secondarticle.title = 'zzzz';

const firstArticle = await createArticle(article);

// send 'bush gehs' as your search query;

now you can assert that the title_scrore === 1


```

Reply:
create a mock that you know will not match the query in any way and assert that it didn't match. That's how you know your search doesn't just match everything in the db

Reply:
All right, we will do that.

Reply:
> create a mock that you know will not match the query in any way and assert that it didn't match. That's how you know your search doesn't just match everything in the db

Okay, we'll implement the changes as requested. 

Thank you